File: cbs-server/src/main/java/com/evil/cbs/domain/Ticket.java
package com.evil.cbs.domain;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.EnumType;
import javax.persistence.Enumerated;
import javax.persistence.ManyToOne;
import javax.persistence.OneToOne;
import javax.persistence.Table;

@Data
@Entity
@EqualsAndHashCode(callSuper = true)
@Table(name = "tickets", schema = "cbs")
@Builder
@AllArgsConstructor(access = AccessLevel.PACKAGE)
@NoArgsConstructor(access = AccessLevel.PACKAGE)
public class Ticket extends AbstractEntity{
    @Column(name = "ticket_status")
    @Enumerated(EnumType.STRING)
    private TicketStatus ticketStatus = TicketStatus.NOT_USED;
    @Column(name = "ticket_type")
    @Enumerated(EnumType.STRING)
    private TicketType ticketType = TicketType.SIMPLE;
    @OneToOne
    private MovieSession movieSession;
    @OneToOne
    private Seat bookedSeat;
    @ManyToOne
    private User user;

}

File: cbs-client/src/app/model/ticket.ts
import {MovieSession} from "./movie-session";
import {Seat} from "./seat";

export class Ticket {
  ticketStatus : string;
  ticketType : string;
  movieSession : MovieSession;
  bookedSeat : Seat;


  constructor(ticketStatus: string, ticketType: string, movieSession: MovieSession, bookedSeat: Seat) {
    this.ticketStatus = ticketStatus;
    this.ticketType = ticketType;
    this.movieSession = movieSession;
    this.bookedSeat = bookedSeat;
  }
}
